src/metabolic_ninja/worker/main.py

```diff
@@ -136,18 +142,31 @@ def main():
     # Register the signal handler
     signal.signal(signal.SIGTERM, functools.partial(on_terminate, channel))
 
-    logger.info("Ready for action, waiting for messages from RabbitMQ")
-    channel.start_consuming()
-
-    logger.info(
-        "Pika consumption loop exited. Cleaning up and terminating application..."
-    )
+    try:
+        logger.info("Ready for action, waiting for messages from RabbitMQ")
+        channel.start_consuming()
 
-    # Wait for any worker threads to complete
-    for thread in worker_threads:
-        thread.join()
+        logger.info(
+            "Pika consumption loop exited. Cleaning up and terminating application..."
+        )
 
-    connection.close()
+        # Wait for any worker threads to complete before closing the connection to
+        # rabbitmq.
+        for thread in worker_threads:
+            thread.join()
+
+        connection.close()
+    except pika.exceptions.ConnectionClosedByBroker:
+        # Connection closed by broker - likely due to redeployment. Exit and rely on the
+        # environment to restart the application. Don't let the exception bubble though;
+        # we don't need a Sentry notification for this.
+        logger.warning("Connection was closed by RabbitMQ broker. Aborting.")
```

Comment:
Suggestion to log the traceback with warning level such that a traceback can be inspected in case there is an unexpected closed connection.

```suggestion
        logger.warning("Connection was closed by RabbitMQ broker. Aborting.", exc_info=error)
```